artifact-app/build.rs

```diff
@@ -23,10 +23,12 @@ lazy_static! {
 fn check_deps() {
     println!("Checking dependencies");
 
+    let path_finding_cmd = if cfg!(windows) {"where"} else {"which"};
```

Comment:
I think I previously had this, and it didn't work in travis. I think travis runs it in a shell-like env in windows.

I'm wondering.... I've heard that windows now supports bash? Does it make sense to build this in windows bash? What version of windows are you building on?

Maybe we could do not just `cfg!(windows)` but also key on the _version_ of windows. Thoughts?

Reply:
There are a few options for "bash" in windows. One is Windows Linux Subsystem, which does give you essentially a true bash shell in windows. This isn't built in though, and requires the system hypervisor. This means that Docker, VirtualBox / other VM environments, and WLS all block each other from running. You can't usually rely on this feature unfortunately - I wish that you could. The other option is the MinGW / Git shell, which is just a program along with a bunch of implementations of common shell utilities. Git shell you can pretty much count on being installed on any windows computer which is git cloning any git repository, it's part of the standard windows git installation. It is used so that git hooks, etc., can run cross-platform. Perhaps this is where Travis runs? It may be possible to enforce running `which` inside of the git shell on windows to accomplish cross-platform support here.

Reply:
fun, I didn't realize all of that.

I think it would be good if windows developers could run -- and it also ran in travis. I'm not sure why bors is having issues and travis isn't running. I have to investigate...